app/experimenter/nimbus-ui/README.md

```diff
@@ -197,6 +166,69 @@ if (message !== "success") {
 console.log("Retrieved record", record);
 ```
 
+## Forms and Validation
+
+**See the form validation section of [Storybook](https://storage.googleapis.com/mozilla-storybooks-experimenter/index.html) for examples and a visual guide to the UX interaction.**
+
+There are three kinds of validation that need to be applied to forms: client-side, server-side, and "required for launch".
+
+### `useCommonForm` and `useCommonNestedForm`
+
+All forms use [`react-bootstrap`](https://react-bootstrap.github.io/components/forms/) as well as the custom convenience hooks `useCommonForm` or `useCommonNestedForm` for ease of use when developing new forms, to promote consistent behavior amongst our forms, and to limit any future changes affecting all forms in limited places. These hooks utilize the [`react-hook-form`](https://react-hook-form.com/api) package.
+
+If there are no nested forms, pass the needed parameters into `useCommonForm`. Otherwise, pull form methods out of _our_ `useForm` hook and pass them into `FormContext`. Then, in the nested form(s), pull out the methods and pass them into `useCommonNestedForm`.
+
+Server errors occur when a user attempts to create or modify a record and the server rejects the request. We surface client-side errors to prevent invalid mutations from being attempted and to give users instant feedback if we know a field is invalid. For non-field server-side errors, refer to the "General Errors" section above.
+
+Both hooks return a `FormErrors` JSX Element, intended for use on every form field, which will display client-side errors, server-side errors, or both, within React Bootstrap's [`Form.Control.Feedback`](https://react-bootstrap.github.io/components/forms/#form-control-feedback-props).
+
+Both hooks also return a form control method that returns an object containing field attributes that should be spread onto the form control or select (`formControlAttrs`).
+
+A basic example of a form field utilizing `useCommonForm`:
+
+```
+<Form.Group controlId="name">
+  <Form.Label>Public name</Form.Label>
+  <Form.Control
+    {...formControlAttrs("name")}
+    type="text"
+  />
+  <FormErrors name="name" />
+</Form.Group>
+```
+
+Form fields are optional by default - that is, users can save pieces of the experiment without filling in every input on the page. To enforce a required field or client-side validation on the field, pass a rule into the method:
```

Comment:
The predefined validators will likely see the most use, but I wonder if we should point out that this argument is actually a [`RegisterOptions`](https://react-hook-form.com/ts#ValidationRulesRef) object, which can take a `validate` function, a `pattern` for regex matching, and a few other handy properties like min, max, etc. In case they want to make a new predefined one, or they need to do something fancy in a specific field.

Reply:
I'll mention something similar and link to the docs, good call out 👍